addons/mediarecorder/userscript.js

```diff
@@ -11,6 +12,39 @@ export default async ({ addon, console, msg }) => {
   let recorder;
   let timeout;
 
+  let recordElem;
+
+  const getStoredOptions = () => {
+    try {
+      return JSON.parse(localStorage.getItem("sa-record-options"));
+    } catch {
+      return {};
+    }
+  };
+
+  const getProjectId = () => {
+    const id = parseInt(window.location.pathname.split("/")[2]);
+    return id ? id : 0;
```

Comment:
```suggestion
    return id ?? 0;
```

Reply:
`parseInt` returns `NaN` when parsing nothing or a string without a number. If I'm correct, using a nullish coalescing operator (??)[^1] here would always return `id` as long as it's not nullish, and NaN is **not nullish**, whereas the conditional operator (? ... : ...) [^2] that we are currently using returns `id` as long as it's _truthy_, and NaN is, of course, falsy.

Is there another way this could be done?

[^1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing
[^2]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator

Reply:
`id || 0`?

Reply:
I should have suggested `||` instead to keep the same behavior, nice catch.